server/src/main/java/org/elasticsearch/search/aggregations/bucket/BestBucketsDeferringCollector.java

```diff
@@ -136,20 +135,12 @@ public void preCollection() throws IOException {
         collector.preCollection();
     }
 
-    @Override
-    public void postCollection() throws IOException {
-        finishLeaf();
-        finished = true;
-    }
-
     /**
      * Replay the wrapped collector, but only on a selection of buckets.
      */
     @Override
     public void prepareSelectedBuckets(long... selectedBuckets) throws IOException {
-        if (finished == false) {
-            throw new IllegalStateException("Cannot replay yet, collection is not finished: postCollect() has not been called");
-        }
+        finishLeaf();
```

Comment:
I'm pretty happy about how this turned out! No more `finished` to check on. It's just all built in now.